tests/features/test_cuda_interface.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import pytest
+import sys
+
+try:
+    import cupy as cp
+except ImportError:
+    pytest.skip("CuPy is required for this test", allow_module_level=True)
+
+from pygenn import genn_model
+
+@pytest.mark.parametrize("precision,np_dtype", [("float", np.float32), ("double", np.float64)])
+def test_cuda_array_interface(make_model, backend, precision, np_dtype):
```

Comment:
there is a mechanism for selecting backends tests run for in https://github.com/genn-team/genn/blob/master/tests/features/conftest.py#L25-L53 which you could extend with a ``backend_cuda`` fixture